src/widget/wtracktableview.cpp

```diff
@@ -663,6 +667,40 @@ void WTrackTableView::slotOpenInFileBrowser() {
     }
 }
 
+void WTrackTableView::slotRemoveFromDisk() {
+       TrackModel* trackModel = getTrackModel();
+    if (!trackModel) {
+        return;
+    }
+    QMessageBox msgBox (QMessageBox::Information, QObject::tr("Remove From Disk"), QObject::tr("Really delete the files from disk? (Permanent Deletion)"));
+    QAbstractButton* pYES = (QAbstractButton*)msgBox.addButton(QMessageBox::Yes);
+    QAbstractButton* pNO  = (QAbstractButton*)msgBox.addButton(QMessageBox::No);
+    QCheckBox alsoHide(QObject::tr("Also hide track in library"), &msgBox);
+    alsoHide.blockSignals(true);
+    msgBox.addButton(&alsoHide, QMessageBox::ActionRole);
+
+    if (QMessageBox::Yes == msgBox.exec())
+    {
+
+
+        QModelIndexList indices = selectionModel()->selectedRows();
+
+        for (const QModelIndex& index : indices) {
+            if (!index.isValid()) {
+              continue;
+            }
+            QString  filenameWithPath = trackModel->getTrackLocation(index);
+            QFile file (filenameWithPath);
+            file.remove();
```

Comment:
Let's handle failure to remove -- keep a list of filenames that we couldn't remove and let the user know somehow...

Reply:
So, two lists, a to do list with the dialog, and a dialog that shows any error, with a list of files which did not get deleted.

Reply:
Could we find out the reason deletion failed?
File doesn't exist? no write access/rights?

The 'Success/Error' window could show either "OK" or a list:
- inexistant files
- failed deletions (try again?)

I realize it might be a not-so-common use case and hard to accomplish, but in case of deletion fails (partly) and the error window was closed, could the track view select the 'failed' tracks?